py3status/modules/external_script.py

```diff
@@ -52,20 +99,55 @@ class Py3status:
     cache_timeout = 15
     format = '{output}'
     localize = True
+    notifications = {}
     script_path = None
     strip_output = False
 
     def post_config_hook(self):
         if not self.script_path:
             raise Exception(STRING_ERROR)
 
+        self.button_refresh = 2
+        self.notification = {'normal': [], 'click': False}
+        if self.notifications:
+            for x in ['changed', 'click', 'current']:
+                self.notification[x] = self.notifications.get(x, False)
+                if self.notification[x]:
+                    if x in ['changed', 'current']:
+                        self.notification['normal'].append(x)
```

Comment:
Still cannot grasp this logic, if notification is added for "changed", we add it to "normal"? What is normal?

Reply:
With only two, it would be just `if current or changed:`.
With only 10 or 100, it would be just `if current or changed or .. or .. or .. or .. False:`

I add `"True"` to the list and we check the list itself to see if it's empty or not... instead of going through the list to find a first `"True"`. When we go in, we now have (already optimized) list.

It's a bit different here because of notifications so we check both `if` statements. In other scenarios, we might use that list to loop through something.

Normal is both `current+changed`, but I don't want to say `current`. Idk.